internal/testing/e2e/fetch_matches_test.go

```diff
@@ -841,3 +842,27 @@ func TestHTTPMMF(t *testing.T) {
 	require.Equal(t, err, io.EOF)
 	require.Nil(t, resp)
 }
+
+// TestUnavailableMMF does a simple test of a profile targeting a gRPC
+// function that's not available.
+func TestUnavailableMMF(t *testing.T) {
+	ctx := context.Background()
+	om := newOM(t)
+
+	stream, err := om.Backend().FetchMatches(ctx, &pb.FetchMatchesRequest{
+		Config: &pb.FunctionConfig{
+			Host: "unavailable",
+			Port: rand.Int31(),
+			Type: pb.FunctionConfig_GRPC,
+		},
+		Profile: &pb.MatchProfile{},
+	})
+	require.Nil(t, err)
+
+	om.SetEvaluator(func(ctx context.Context, in <-chan *pb.Match, out chan<- string) error {
+		return nil
+	})
+
+	_, err = stream.Recv()
+	require.Error(t, err)
```

Comment:
Also check that the call takes more than rpcConnectionTimeout but less than registrationInterval + proposalCollectionInterval.

Reply:
I worry that by adding checks like this we'll probably get some timing issues.

Reply:
We've done it in several places without issue.